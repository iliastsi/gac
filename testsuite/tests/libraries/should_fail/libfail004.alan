string():proc
  s : byte[42];
  i : int;
{
  i = strlen(42);
  i = strlen(s, 42);
  s = strlen(s);

  i = strcmp(s, i);
  s = strcmp(s, s);

  strcpy(i, s);
  strcpy(i, i);
  strcpy(i, i, s);

  i = strcat(s, i);
}
