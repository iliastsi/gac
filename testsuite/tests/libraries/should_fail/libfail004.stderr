libfail004.alan:5:14: Error: Type mismatch at `42'
	Incompatible type of argument 1 of function `strlen'
	Expected `array of byte' but argument is of type `int'

libfail004.alan:6:7: Error: Type mismatch at `strlen(s, 42)'
	The function `strlen' is applied to 2 parameters but its type has 1

libfail004.alan:7:3: Error: Type mismatch at `s = strlen(s);'
	Lvalue is of type `array of byte' but Rvalue is of type `int'

libfail004.alan:9:17: Error: Type mismatch at `i'
	Incompatible type of argument 2 of function `strcmp'
	Expected `array of byte' but argument is of type `int'

libfail004.alan:10:3: Error: Type mismatch at `s = strcmp(s, s);'
	Lvalue is of type `array of byte' but Rvalue is of type `int'

libfail004.alan:12:10: Error: Type mismatch at `i'
	Incompatible type of argument 1 of function `strcpy'
	Expected `array of byte' but argument is of type `int'

libfail004.alan:13:10: Error: Type mismatch at `i'
	Incompatible type of argument 1 of function `strcpy'
	Expected `array of byte' but argument is of type `int'

libfail004.alan:13:13: Error: Type mismatch at `i'
	Incompatible type of argument 2 of function `strcpy'
	Expected `array of byte' but argument is of type `int'

libfail004.alan:14:3: Error: Type mismatch at `strcpy(i, i, s)'
	The function `strcpy' is applied to 3 parameters but its type has 2

libfail004.alan:16:3: Error: Type mismatch at `i = strcat(s, i);'
	Lvalue is of type `int' but Rvalue is of type `proc'

libfail004.alan:16:17: Error: Type mismatch at `i'
	Incompatible type of argument 2 of function `strcat'
	Expected `array of byte' but argument is of type `int'

